sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -242,25 +221,19 @@ first_seen_date AS (
   GROUP BY
     client_id
 ),
--- The next CTE returns the second_seen_date. It finds the next date after first_seen_date
--- as reported by the main ping. Further dates reported by other pings are skipped.
--- The source ping for second_seen_date is always the main ping.
+-- The next CTE returns the second_seen_date calculated as the next date reported by the
+--  main ping after first_seen_date or NULL. Dates reported by other pings are excluded.
 second_seen_date AS (
-  SELECT
+   SELECT
     client_id,
-    IF(
-      ARRAY_LENGTH(ARRAY_AGG(seen_dates)) > 1,
-      ARRAY_AGG(seen_dates ORDER BY value_date ASC)[SAFE_OFFSET(1)],
-      NULL
-    ) AS second_seen_date
+    all_dates[SAFE_OFFSET(0)] AS second_seen_date
```

Comment:
And what if the first date is the same as the first date of the main ping? Shouldn't we be capturing offset 1 here?

Reply:
That would happen is there was another ping type before, so it's okay to retrieve the first main ping reported as long as it's after the `first_seen_date`.